homeassistant/components/stream/hls.py

```diff
@@ -109,14 +137,67 @@ def name(self) -> str:
     @property
     def format(self) -> str:
         """Return container format."""
-        return "mpegts"
+        return "mp4"
 
     @property
     def audio_codec(self) -> str:
         """Return desired audio codec."""
         return "aac"
 
     @property
-    def video_codec(self) -> str:
-        """Return desired video codec."""
-        return "h264"
+    def video_codecs(self) -> tuple:
+        """Return desired video codecs."""
+        return {"hevc", "h264"}
+
+    @property
+    def container_options(self) -> dict:
+        """Return container options."""
+        return {"movflags": "empty_moov+default_base_moof"}
+
+
+def find_box(segment: io.BytesIO, target_type: bytes, box_start: int = 0) -> int:
```

Comment:
Can you walk me through the need for these methods? Is there a better place for them to live? Perhaps a utils file?

Reply:
This tools are needed to edit the fmp4 boxes. We could move them to another file...